pkg/admissioncontroller/webhooks/admission/namespacedeletion/admission.go

```diff
@@ -36,109 +36,107 @@ import (
 )
 
 const (
-	namespaceValidatorName = "namespace_validator"
+	// PluginName is the name of this admission plugin.
+	PluginName = "namespace_validator"
+	// WebhookPath is the HTTP handler path for this admission webhook handler.
+	WebhookPath = "/webhooks/validate-namespace-deletion"
 )
 
 var namespaceGVK = metav1.GroupVersionKind{Group: "", Kind: "Namespace", Version: "v1"}
 
-type namespaceDeletionHandler struct {
-	cacheReader client.Reader
-	apiReader   client.Reader
-
-	logger logr.Logger
-}
-
-// NewValidateNamespaceDeletionHandler creates a new handler for validating namespace deletions.
-func NewValidateNamespaceDeletionHandler(ctx context.Context, cache cache.Cache) (admission.Handler, error) {
+// New creates a new handler for validating namespace deletions.
+func New(ctx context.Context, logger logr.Logger, cache cache.Cache) (admission.Handler, error) {
 	// Initialize caches here to ensure the readyz informer check will only succeed once informers required for this
 	// handler have synced so that http requests can be served quicker with pre-syncronized caches.
-	_, err := cache.GetInformer(ctx, &gardencorev1beta1.Project{})
-	if err != nil {
+	if _, err := cache.GetInformer(ctx, &gardencorev1beta1.Project{}); err != nil {
 		return nil, err
 	}
 
-	return &namespaceDeletionHandler{
+	return &plugin{
 		cacheReader: cache,
+		logger:      logger,
 	}, nil
 }
 
-// InjectLogger injects a logger into the handler.
-func (h *namespaceDeletionHandler) InjectLogger(l logr.Logger) error {
-	h.logger = l.WithName(namespaceValidatorName)
-	return nil
+type plugin struct {
```

Comment:
Naming nit: `plugin` might be confusing, as this is not an admission plugin. Maybe `handler` fits better?

Reply:
Yes, makes sense, done in d80c0c5